### PR TITLE
crypto: fixes & remove AES-GCM

### PR DESCRIPTION
**Pull request by enkore**
Fixups for #1034

### CODE REVIEW

File: src/borg/crypto/low_level.pyx [hunk 1]
```diff
@@ -67,7 +67,6 @@ cdef extern from "openssl/evp.h":
         pass
 
     const EVP_CIPHER *EVP_aes_256_ctr()
-    const EVP_CIPHER *EVP_aes_256_gcm()
```
Comment (ThomasWaldmann) [inline src/borg/crypto/low_level.pyx]: i intentionally did not remove gcm although some people have bad feelings about it. other people have also bad feelings about patents (even with the patent license for FOSS and openssl), so there would be gcm as option for them.

technically, I'ld also choose ocb over gcm.

File: src/borg/crypto/low_level.pyx [hunk 2]
```diff
@@ -223,8 +222,8 @@ cdef class AES256_CTR_BASE:
     cdef unsigned char iv[16]
     cdef long long blocks
 
-    @staticmethod
-    def requirements_check():
+    @classmethod
+    def requirements_check(cls):
```
Comment (ThomasWaldmann) [inline src/borg/crypto/low_level.pyx]: interesting, i didn't notice malfunction here. did you file a bug at cython?

Comment (enkore) [inline src/borg/crypto/low_level.pyx]: It's not a malfunction, just slightly incorrect code.

```
foo something(); // this is not a prototype for the function below

foo
something() { // this function was not declared above!
 return ...;
}
```

File: src/borg/crypto/low_level.pyx (hunk 1 above)
Comment (enkore) [inline src/borg/crypto/low_level.pyx]: chapoly > either, like I argued previously. With no performance impact.

Comment (enkore) [inline src/borg/crypto/low_level.pyx]: See https://github.com/borgbackup/borg/issues/1031#issuecomment-304049587

Comment (enkore) [inline src/borg/crypto/low_level.pyx]: So, remove OCB as well.?

Comment (ThomasWaldmann) [inline src/borg/crypto/low_level.pyx]: no, keep it, it is hw accelerated and fastest.

**Pull request merged**